storm/src/test/storm/mesos/schedulers/DefaultSchedulerTest.java

```diff
@@ -295,25 +298,47 @@ public void testAllSlotsAvailableForSchedulingWithMultipleOffers() {
     offers.add(buildOfferWithPorts("offer6", sampleHost, 0, 0, samplePort, samplePort + 5));
     offers.add(buildOffer("offer7", sampleHost, 4 * DEFAULT_WORKER_CPU, 0));
     offers.add(buildOffer("offer8", sampleHost, 0, 4 * DEFAULT_WORKER_MEM));
-
-    offers.add(buildOffer("offer3", sampleHost2, DEFAULT_EXECUTOR_CPU + DEFAULT_WORKER_CPU, 0));
-    offers.add(buildOffer("offer4", sampleHost2, 0, DEFAULT_EXECUTOR_MEM + DEFAULT_WORKER_MEM));
-    offers.add(buildOffer("offer5", sampleHost2, 0.01, 10));
-    offers.add(buildOfferWithPorts("offer9", sampleHost2, 0, 0, samplePort, samplePort));
+    /*
+     * XXX(erikdw): add a hacky fudge-factor of 0.01 for now, to allow the floating-point
+     * tabulation logic in AggregatedOffers to succeed.  We will switch to fixed-point math
+     * later to allow exact calculations to succeed.
+     * See this issue for more info: https://github.com/mesos/storm/issues/161
+     *
+     * Once that is fixed we can remove this 0.01 fudge-factor line, and we should also
+     * analyze other places in this project's tests that have y.x1 (e.g., 0.01, 0.91, etc.).
+     * i.e.,
+     *
+     *  % git grep -E '\..1' | grep -v pom.xml
+     *  storm/src/main/storm/mesos/util/MesosCommon.java:  public static final double MESOS_MIN_CPU = 0.01;
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    assertEquals(0.4f, TestUtils.calculateAllAvailableScalarResources(aggregatedOffersPerNode.get("h1"), ResourceType.CPU), 0.01f);
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    assertEquals(100f, TestUtils.calculateAllAvailableScalarResources(aggregatedOffersPerNode.get("h1"), ResourceType.MEM), 0.01f);
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    assertEquals(TestUtils.calculateAllAvailableScalarResources(aggregatedOffersPerNode.get("h1"), ResourceType.CPU), 0.4f, 0.01f);
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    assertEquals(TestUtils.calculateAllAvailableScalarResources(aggregatedOffersPerNode.get("h1"), ResourceType.MEM), 100f, 0.01f);
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    offer = TestUtils.buildOffer("O-H1-2", "h1", 3.21, 0);
+     *  storm/src/test/storm/mesos/MesosNimbusTest.java:    offer = TestUtils.buildOffer("O-H2-2", "h2", 3.21, 0);
+     *  storm/src/test/storm/mesos/schedulers/DefaultSchedulerTest.java:     * all of this project's tests that have y.x1 (e.g., 0.01, 0.91, etc.).
+     *  storm/src/test/storm/mesos/schedulers/DefaultSchedulerTest.java:    offers.add(buildOffer("offer9", sampleHost, 0.01, 0));
+     *  storm/src/test/storm/mesos/schedulers/DefaultSchedulerTest.java:    offers.add(buildOffer("offer12", sampleHost2, 0.01, 10));
+     */
+    offers.add(buildOffer("offer9", sampleHost, 0.01, 0));
```

Comment:
So this hacky fudge-factor is only in lines 323 and 327, correct? If so, can we re-arrange some of these offers so that the non-hacky offers are all above this giant comment block, and only the hacky ones appear below it?